psydac/api/fem.py

```diff
@@ -715,17 +763,50 @@ def allocate_matrices(self, backend=None):
 
 #==============================================================================
 class DiscreteLinearForm(BasicDiscrete):
+    """ Class that represents the concept of a discrete linear form.
+        This class allocate the vector and generates the vector assembly method.
+
+    Parameters
+    ----------
+
+    expr : sympde.expr.expr.LinearForm
+        The symbolic bi-linear form.
+
+    kernel_expr : sympde.expr.evaluation.KernelExpression
+        The atomic representation of the linear form.
+
+    domain_h : Geometry
+        The discretized domain
+
+    space : FemSpace
+        The discrete spaces.
+
+    vector : Vector
+        The vector that we assemble into it.
```

Comment:
I would add: if not provided, the assembly method will create a new `Vector` of the appropriate space.

Reply:
done in [9ed13cb](https://github.com/pyccel/psydac/pull/232/commits/9ed13cb7ef3cc0b2cf5ef6284c62b95ffd2874f1)